clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -61,6 +82,9 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent {
   CacheResult loadData(ProviderCache providerCache) {
     CacheResultBuilder cacheResultBuilder = new CacheResultBuilder(startTime: Long.MAX_VALUE)
 
+    // Lookup all instances and group by stack Id
+    Map<String, List<Server>> instancesByStackId = clientProvider.getInstances(region)?.groupBy { Server server -> server?.metadata['metering.stack'] ?: 'unknown' }
```

Comment:
We might want to abstract out our strategy for finding servers associated to stacks - metering.stack was something thrown in to the template from the github hot templates, but not sure that's how we will always mark servers.